src/binaries/query/main.rs

```diff
@@ -251,3 +257,14 @@ fn run_cmd(conf: &Config) -> bool {
 
     true
 }
+
+#[cfg(not(target_os = "macos"))]
+fn try_set_max_open_files() {
+    let limits = get_own_limits().unwrap();
```

Comment:
`get_own_limits` could return errors. If unwrap here, databend-query will fail to start. How about printing a message instead?

For example:

```suggestion
    let limits = match get_own_limits() {
        Ok(limits) => limits,
        Err(err) => {
            warn!("get system limit of databend-query failed: {e:?}";
            return
        }
    }
```